crates/router/src/connector/mollie/transformers.rs

```diff
@@ -0,0 +1,383 @@
+                        api_models::payments::BankRedirectData::Eps { .. } => PaymentMethod::Eps,
+                        api_models::payments::BankRedirectData::Giropay { .. } => {
+                            PaymentMethod::Giropay
+                        }
+                        api_models::payments::BankRedirectData::Ideal { .. } => {
+                            PaymentMethod::Ideal
+                        }
+                        api_models::payments::BankRedirectData::Sofort { .. } => {
+                            PaymentMethod::Sofort
+                        }
+                    };
+                    Ok(payment_method)
+                }
+                _ => Err(errors::ConnectorError::NotImplemented(
+                    "Payment Method".to_string(),
+                )),
+            },
+            _ => {
+                let flow: String = format!(
+                    "{} capture",
+                    item.request.capture_method.unwrap_or_default()
+                );
+                Err(errors::ConnectorError::FlowNotSupported {
+                    flow,
+                    connector: "Mollie".to_string(),
+                })
+            }
+        }?;
+        let billing_address = get_billing_details(item)?;
+        let shipping_address = get_shipping_details(item)?;
+        Ok(Self {
+            amount,
+            description,
+            redirect_url,
+            cancel_url: None,
+            webhook_url: "".to_string(),
+            locale: None,
+            method,
+            metadata: None,
+            sequence_type: SequenceType::Oneoff,
+            mandate_id: None,
+            shipping_address,
+            billing_address,
+            card_token: None,
+            // To do if possible this should be from the payment request
+            issuer: None,
+        })
+    }
+}
+
+fn get_shipping_details(
+    item: &types::PaymentsAuthorizeRouterData,
+) -> Result<Option<Address>, error_stack::Report<errors::ConnectorError>> {
+    let shipping_address = item
+        .address
+        .shipping
+        .as_ref()
+        .and_then(|shipping| shipping.address.as_ref());
+    get_address_details(shipping_address)
+}
+
+fn get_billing_details(
+    item: &types::PaymentsAuthorizeRouterData,
+) -> Result<Option<Address>, error_stack::Report<errors::ConnectorError>> {
+    let billing_address = item
+        .address
+        .billing
+        .as_ref()
+        .and_then(|billing| billing.address.as_ref());
+    get_address_details(billing_address)
+}
+
+fn get_address_details(
+    address: Option<&AddressDetails>,
+) -> Result<Option<Address>, error_stack::Report<errors::ConnectorError>> {
+    let address_details = match address {
+        Some(address) => {
+            let street_and_number = Secret::new(format!(
+                "{},{}",
+                address.get_line1()?.peek(),
+                address.get_line2()?.peek()
+            ));
```

Comment:
move this `street_and_number ` to connector utils, so that we don't accidentally expose the PII data